model-engine/model_engine_server/infra/services/live_llm_batch_completions_service.py

```diff
@@ -0,0 +1,81 @@
+from datetime import datetime, timedelta
+from typing import Dict, Optional
+
+from model_engine_server.common.dtos.batch_jobs import CreateDockerImageBatchJobResourceRequests
+from model_engine_server.common.dtos.llms import (
+    BatchCompletionsJob,
+    BatchCompletionsJobStatus,
+    CreateBatchCompletionsEngineRequest,
+)
+from model_engine_server.core.auth.authentication_repository import User
+from model_engine_server.domain.gateways.docker_image_batch_job_gateway import (
+    DockerImageBatchJobGateway,
+)
+from model_engine_server.domain.services.llm_batch_completions_service import (
+    LLMBatchCompletionsService,
+)
+
+
+class LiveLLMBatchCompletionsService(LLMBatchCompletionsService):
+    def __init__(
+        self,
+        docker_image_batch_job_gateway: DockerImageBatchJobGateway,
+    ):
+        self.docker_image_batch_job_gateway = docker_image_batch_job_gateway
+
+    async def create_batch_job(
+        self,
+        *,
+        user: User,
+        image_repo: str,
+        image_tag: str,
+        job_request: CreateBatchCompletionsEngineRequest,
+        resource_requests: CreateDockerImageBatchJobResourceRequests,
+        max_runtime_sec: int = 24 * 60 * 60,
+        labels: Dict[str, str] = {},
+        priority: Optional[int] = 0,
+        num_workers: Optional[int] = 1,
```

Comment:
note: when multinode comes we're probably gonna need to figure out different semantics for num_workers. nothing needs to done now, just want to flag

Reply:
just ported this over, but agreed we can remove/replace when we get to multinode